compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -4622,6 +4622,19 @@ public boolean isUnionOfSimpleBasicTypes(BType type) {
         return isSimpleBasicType(type.tag);
     }
 
+    public boolean isUnionOfSimpleBasicTypesExceptNilType(BType type) {
```

Comment:
I think the accurate name now would be something like
```suggestion
    public boolean isNonNilSimpleBasicTypeOrString(BType type) {
```